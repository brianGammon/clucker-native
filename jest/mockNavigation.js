/* @flow */
type ReturnVals = {
  [methodName: string]: {
    [param: string]: string,
  },
};

type MockNavigation = {
  getParam: (param: string, defaultId: string) => string,
};

const mockNavigation = (returnVals: ReturnVals) => ({
  getParam: jest.fn((param: string, defaultId: string) => {
    if (returnVals.getParam && returnVals.getParam[param]) {
      return returnVals.getParam[param];
    }
    return defaultId;
  }),
}: MockNavigation);

export default mockNavigation;
