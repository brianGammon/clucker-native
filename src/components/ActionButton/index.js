import React from 'react';
import moment from 'moment';
import { Button, Icon, ActionSheet } from 'native-base';
import { type Navigation } from '../../types';
import { nowAsMoment } from '../../utils/dateHelper';

type Props = {
  navigation: Navigation,
};

class ActionButton extends React.Component<Props> {
  handlePress = () => {
    const { navigation } = this.props;
    const chickenId = navigation.getParam('chickenId', null);
    const options = {};
    if (chickenId) {
      options.chickenId = chickenId;
    }
    let date = navigation.getParam('date', null);
    if (date) {
      if (date.length === 7) {
        const now = nowAsMoment();
        if (now.month() === moment.utc(date).month()) {
          date = now.format('YYYY-MM-DD');
        } else {
          date += '-01';
        }
      }
      options.date = date;
    }
    const BUTTONS = ['Log an egg', 'Bulk egg entry', 'Add a chicken', 'Cancel'];
    const ACTIONS = [
      () => navigation.navigate('EggEditor', { ...options }),
      () => navigation.navigate('BulkEditor', { ...options }),
      () => navigation.navigate('ChickenEditor'),
      () => {},
    ];
    const CANCEL_INDEX = 3;
    ActionSheet.show(
      {
        options: BUTTONS,
        cancelButtonIndex: CANCEL_INDEX,
        title: 'Select an action:',
      },
      (buttonIndex) => {
        ACTIONS[buttonIndex]();
      },
    );
  };

  render() {
    return (
      <Button transparent onPress={this.handlePress}>
        <Icon active name="add-circle" />
      </Button>
    );
  }
}

export default ActionButton;
