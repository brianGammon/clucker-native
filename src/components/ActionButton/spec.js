import React from 'react';
import mockNavigation from '../../../jest/mockNavigation';
import ActionButton from '.';

describe('ActionButton', () => {
  const navigation = mockNavigation({
    getParams: {
      chickenId: 'chicken1',
      date: '2018-08-05',
    },
  });
  test('renders', () => {
    const wrapper = shallow(<ActionButton navigation={navigation} />);
    expect(wrapper).toMatchSnapshot();
  });
});
