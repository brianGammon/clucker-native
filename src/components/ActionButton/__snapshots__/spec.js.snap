// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton renders 1`] = `
<Styled(Button)
  onPress={[Function]}
  transparent={true}
>
  <Styled(Icon)
    active={true}
    name="add-circle"
  />
</Styled(Button)>
`;
